client_test.go

```diff
@@ -0,0 +1,57 @@
+package osrm
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_getWithError(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(500)
+		fmt.Fprint(w, "something wrong")
+	}))
+	defer ts.Close()
+
+	c := newClient("/", ts.Client())
+	b, err := c.get(context.Background(), "/")
+	require.Nil(t, b)
+	require.NotNil(t, err)
+}
+
+func Test_doRequestWithBadHTTPCode(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(500)
+		fmt.Fprint(w, "<html><head>")
+	}))
+	defer ts.Close()
+
+	c := newClient(ts.URL, ts.Client())
+	req := request{
+		profile: "something",
+		geoPath: geoPath,
+		service: "foobar",
+	}
+	err := c.doRequest(context.Background(), &req, nil)
+	require.EqualError(t, err, "unexpected http status code 500 with body \"<html><head>\"")
+}
+
+func Test_doRequestWithBodyUnmarshalFailure(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+	}))
+	defer ts.Close()
+
+	c := newClient(ts.URL, ts.Client())
```

Comment:
Ohhhh, turns out this `.Client()` stuff breaks go 1.8 support.. Maybe that was a bad idea..

Reply:
Actually I believe this `ts.Client()` should be probably replaceable with just `&http.Client{}` (the only thing we really care about is `ts.URL`. Could you please check if that's true?

For me it seems preferable to keep go1.8 compatibilty (even though it affects tests only).